otj-metrics-core/src/main/java/com/opentable/metrics/common/SortedEntry.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.metrics.common;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ComparisonChain;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+
+import com.opentable.metrics.http.HealthController;
+import com.opentable.metrics.ready.ReadyCheck;
+import com.opentable.metrics.ready.ReadyController;
+
+/**
```

Comment:
Common grouping and rendering code for both health and ready checks used by controllers

Reply:
This should be the JavaDoc. The class name is kind of vauge, so a doc is in order.